analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -451,5 +451,301 @@ public void Finally_NestedFinally()
+try
+{
+    Tag(""InTry"");
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""End"";";
+            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",
+                "End");
+        }
+
+        [TestMethod]
+        public void Catch_Simple_TypeFilter()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+}
+catch (Exception)
+{
+    tag = ""InCatch"";
+}
+tag = ""End"";";
+            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",
+                "End");
+        }
+
+        [TestMethod]
+        public void Catch_Multiple()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+}
+catch (ArgumentNullException ex)
+{
+    tag = ""InCatchArgumentNull"";
+}
+catch (NotSupportedException ex)
+{
+    tag = ""InCatchNotSupported"";
+}
+catch (Exception ex)
+{
+    tag = ""InCatchEverything"";
+}
+tag = ""End"";";
+            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "End",
+                "InCatchArgumentNull",
+                "InCatchNotSupported",
+                "InCatchEverything");
+        }
+
+        [TestMethod]
+        public void Catch_Finally()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+}
+catch (Exception ex)
+{
+    tag = ""InCatch"";
+}
+finally
+{
+    tag = ""InFinally"";
+}
+tag = ""End"";";
+            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InFinally",    // Happy path
+                "InCatch",      // Exception thrown by Tag("InTry")
+                "End");
+        }
+
+        [TestMethod]
+        public void Catch_NestedInTry()
+        {
+            const string code = @"
+var tag = ""BeforeOuterTry"";
+try
+{
+    Tag(""BeforeInnerTry"");    // Can throw
+    try
+    {
+        Tag(""InInnerTry"");    // Can throw
+    }
+    catch (Exception exInner)
+    {
+        tag = ""InInnerCatch"";
+    }
+    tag = ""AfterInnerTry"";
+}
+catch (Exception ex)
+{
+    tag = ""InOuterCatch"";
+}
+tag = ""End"";";
+            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+                "BeforeOuterTry",
+                "BeforeInnerTry",
+                "InOuterCatch",
+                "InInnerTry",
+                "End",
+                "AfterInnerTry",
+                "InInnerCatch");
+        }
+
+        [TestMethod]
+        public void Catch_NestedInCatch()
+        {
+            const string code = @"
+var tag = ""BeforeOuterTry"";
+try
+{
+    Tag(""InOuterTry"");
+}
+catch (Exception ex)
+{
+    tag = ""BeforeInnerTry"";
+    try
+    {
+        Tag(""InInnerTry"");
```

Comment:
I think we should replace these with throws after I merge my PR, it will make the intent obvious

Reply:
The throw functionality is not available yet

Reply:
now? :)

Reply:
:disappear:

Reply:
I would have to tag & throw to be able to assert it.